blockchain/chain/src/store/chain_store.rs

```diff
@@ -501,7 +501,7 @@ where
             //      or by refactoring `walk_snapshot` to take an async callback.
             // * If cb can return a generic type, deserializing would remove need to clone.
             // Ignore error intentionally, if receiver dropped, error will be handled below
-            let _ = task::block_on(tx.send((cid, block.clone())));
+            let _ = tokio::runtime::Handle::current().block_on(tx.send((cid, block.clone())));
```

Comment:
I would feel safer if you passed in the `Handle` from the outside.

Reply:
Oh and also update the comment above the code.

Reply:
Fixed